go/engine/paperprovision.go

```diff
@@ -54,9 +54,7 @@ func (e *PaperProvisionEngine) Run(m libkb.MetaContext) (err error) {
 	defer m.Trace("PaperProvisionEngine#Run", func() error { return err })()
 
 	// clear out any existing session
-	// XXX THIS IS DANGEROUS SINCE IT COULD CATASTROPHICALLY DELETE A NOPW KEYCHAIN ENTRY XXX
-	// XXX FIXME XXX
-	err = m.Logout()
+	err = m.LogoutKeepSecrets()
```

Comment:
not sure?

Reply:
I don't think you should be able to land here if you are logged in. Maybe it's better to bail out if we are logged in (so we are in bad state from this engine's point of view) instead of logging out to clear the state.

Reply:
Doesn't seem bad to me if you're logged in when you do this. We can just say provisioning a device is always done with login -s.

Reply:
My questions is more like what exactly happens when you are user switching? doesn't it first logs the current user out (with keepSecrets)? So is the only expected state to be in this engine is logged out?